bakery/bakery.go

```diff
@@ -0,0 +1,99 @@
+package bakery
+
+import (
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
+)
+
+// Bakery is a convenience type that contains both an Oven
+// and a Checker.
+type Bakery struct {
+	Oven    *Oven
+	Checker *Checker
+}
+
+// BakeryParams holds a selection of parameters for the Oven
+// and the Checker created by New.
+//
+// For more fine-grained control of parameters, create the
+// Oven or Checker directly.
+//
+// The zero value is OK to use, but won't allow any authentication
+// or third party caveats to be added.
+type BakeryParams struct {
+	// Checker holds the checker used to check first party caveats.
+	// If this is nil, New will use checkers.New(nil).
+	Checker FirstPartyCaveatChecker
+
+	// RootKeyStore holds the root key store to use. If you need to
+	// use a different root key store for different operations,
+	// you'll need to pass a RootKeyStoreForOps value to NewOven
+	// directly.
+	//
+	// If this is nil, New will use NewMemRootKeyStore().
```

Comment:
a statement saying that for production, this should never be nil.. or something to that effect..